tests/TestCase/Console/ConsoleOptionParserTest.php

```diff
@@ -185,6 +185,60 @@ public function testAddOptionDefault(): void
         $this->assertEquals(['test' => 'default value', 'help' => false], $result[0], 'Default value did not parse out');
     }
 
+    /**
+     * test adding an option with a non-string default.
+     */
+    public function testAddOptionIntegerDefault(): void
+    {
+        $parser = new ConsoleOptionParser('test', false);
+        $parser
+            ->addOption('test', [
+                'default' => 1,
+            ])
+            ->addOption('no-default', []);
+
+        $result = $parser->parse(['--test'], $this->io);
+        $this->assertSame(
+            ['test' => '1', 'help' => false],
+            $result[0],
+            'Default value did not parse out',
+        );
+
+        $parser = new ConsoleOptionParser('test', false);
+        $parser->addOption('test', [
+            'default' => 1,
+        ]);
+        $result = $parser->parse([], $this->io);
+        $this->assertEquals(['test' => '1', 'help' => false], $result[0], 'Default value did not parse out');
+    }
+
+     /**
+     * test adding an option with a non-string default.
+     */
+    public function testAddOptionBooleanDefault(): void
+    {
+        $parser = new ConsoleOptionParser('test', false);
+        $parser
+            ->addOption('test', [
+                'default' => true,
```

Comment:
This would be a breaking change for many applications, as you would have been able to get booleans from `--enabled` style options. I don't think we should be casting bools to strings.